src/server/server.rs

```diff
@@ -344,6 +394,35 @@ where
         Ok(())
     }
 
+    pub fn pause(&mut self) -> Result<()> {
+        info!("pausing the grpc server");
```

Comment:
Taining's point is valid. It's better to add a time watch on this function. Same for the resume.